readme.md

```diff
@@ -910,13 +910,20 @@ Assert that `value` is deeply equal to `expected`. See [Concordance](https://git
 
 Assert that `value` is not deeply equal to `expected`. The inverse of `.deepEqual()`.
 
-### `.throws(function|promise, [error, [message]])`
+### `.throws(thrower, [expected, [message]])`
 
-Assert that `function` throws an error, `promise` rejects with an error, or `function` returns a rejected `promise`.
+Assert that an error is thrown. `thrower` can be a function which should throw, or return a promise that should reject, or an observable that should error. Alternatively a promise or observable can be passed directly.
 
-`error` can be an error constructor, error message, regex matched against the error message, or validation function.
+The thrown value *must* be an error. It is returned so you can run more assertions against it.
 
-Returns the error thrown by `function` or a promise for the rejection reason of the specified `promise`.
+`expected` can be a constructor, in which case the thrown error must be an instance. It can be a string, which is compared against the thrown error's message, or a regular expression which is matched against this message. You can also specify a matcher object with one or more of the following properties:
+
+* `is`: the thrown error must be strictly equal to `expected.is`
+* `message`: either a string, which is compared against the thrown error's message, or a regular expression, which is matched against this message
+* `name`: the expected `.name` value of the thrown error
+* `of`: a constructor, the thrown error must be an instance of
+
+`expected` does not need to be specified. If you don't need it but do want to set an assertion message you have to specify `null` or the empty object `{}`.
```

Comment:
Since we're being strict, can we only support `null`? Doesn't make sense to specify an empty object.